lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -183,6 +183,7 @@ def new_transaction_id
       def commit(request)
         url = (test? ? test_url : live_url)
 
+        # ssl_post raises an exception for any non-2xx HTTP status from the gateway
         begin
```

Comment:
Consider refactoring the `commit` method to use a rescue statement similar to this example: https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/kushki.rb#L137

By bringing more logic into the `rescue` statement, you should be able to delete the `response_from_request_error` since it's so similar to the response statement contained inside `commit`.

Reply:
I like that - good find. Done.